pkg/workload/pgx_helpers.go

```diff
@@ -319,32 +315,57 @@ func (m *MultiConnPool) WarmupConns(ctx context.Context, numConns int) error {
 	// (128).
 	g.SetLimit(100)
 
-	var warmupConnsPerPool []int
+	type warmupPool struct {
+		maxConns int
+		pool     *pgxpool.Pool
+	}
+	warmupPools := make([]warmupPool, len(m.Pools))
 	if numConns == 0 {
-		warmupConnsPerPool = make([]int, len(m.Pools))
 		for i, p := range m.Pools {
-			warmupConnsPerPool[i] = int(p.Config().MaxConns)
+			warmupPools[i].maxConns = int(p.Config().MaxConns)
+			warmupPools[i].pool = p
 		}
 	} else {
-		warmupConnsPerPool = distribute(numConns, len(m.Pools))
+		numPoolConns := distribute(numConns, len(m.Pools))
 		for i, p := range m.Pools {
-			poolMaxConns := int(p.Config().MaxConns)
-			if warmupConnsPerPool[i] > poolMaxConns {
-				warmupConnsPerPool[i] = poolMaxConns
+			warmupPools[i].pool = p
+			warmupPools[i].maxConns = numPoolConns[i]
+			maxPoolConns := int(p.Config().MaxConns)
+			if warmupPools[i].maxConns > maxPoolConns {
+				warmupPools[i].maxConns = maxPoolConns
 			}
 		}
 	}
 
 	var numWarmupConns int
-	for _, n := range warmupConnsPerPool {
-		numWarmupConns += n
+	for i, p := range warmupPools {
+		numWarmupConns += p.maxConns
+		if p.maxConns <= 0 {
+			warmupPools[i].maxConns = 1
```

Comment:
Can we fix this up in the previous for loop? As is, I'm confused by why we add `p.maxConns` to `numWarmupConns` *before* then overwriting `p.maxConns` with 1. I also don't understand why `distribute` would ever return a zero for a pool. Can we not fold all of these semantics into `distribute` and simplify the code here?